website/addons/mendeley/model.py

```diff
@@ -0,0 +1,270 @@
+import time
+
+import mendeley
+from modularodm import fields
+
+
+from website import settings
+from website.addons.base import AddonNodeSettingsBase
+from website.addons.base import AddonUserSettingsBase
+from website.citations.models import Citation
+from website.citations.models import CitationList
+from website.oauth.models import ExternalProvider
+
+from .api import APISession
+
+
+class AddonMendeleyUserSettings(AddonUserSettingsBase):
+
+    def _get_connected_accounts(self):
+        """Get user's connected Mendeley accounts"""
+        return [
+            x for x in self.owner.external_accounts if x.provider == 'mendeley'
+        ]
+
+    def to_json(self, user):
+        rv = super(AddonMendeleyUserSettings, self).to_json(user)
+        rv['accounts'] = [
+            {
+                'id': account._id,
+                'provider_id': account.provider_id,
+                'display_name': account.display_name,
+            } for account in self._get_connected_accounts()
+        ]
+        return rv
+
+
+class AddonMendeleyNodeSettings(AddonNodeSettingsBase):
+    external_account = fields.ForeignField('externalaccount',
+                                           backref='connected')
+
+    mendeley_list_id = fields.StringField()
+
+    # Keep track of all user settings that have been associated with this
+    #   instance. This is so OAuth grants can be checked, even if the grant is
+    #   not currently being used.
+    associated_user_settings = fields.AbstractForeignField(list=True)
```

Comment:
Why is this an abstract field? Can it reference things that aren't `AddonMendeleyUserSettings` records?